aten/src/ATen/core/jit_type.h

```diff
@@ -782,6 +782,19 @@ struct CAFFE2_API NamedType : public Type {
   c10::optional<QualifiedName> name_;
 };
 
+// Any should never appear in a named type like a class, namedtuple or
+// interface. If it does, then dynamic type information will be lost in the
+// Pickler, leading to hard-to-track-down bugs that will only occur
+// after saving or loading a model. This is because we rely on the
+// static types in named types to reconstruct type tags of loaded
+// values. Lifting this restriction requires solving the serialization
+// problem first.
+CAFFE2_API void checkNoAny(
```

Comment:
I am a little wary of having this enforcement as opt-in. For example, if someone goes in and adds attributes to interfaces and forgets to include this check, we've very quickly produced the bug this comment describes.

Reply:
The specific example of interfaces will not actually cause this bug, but I agree in spirit. But how would we make it more likely to be called? There isn't a more common place to put it.